gpkit/constraints/sgp.py

```diff
@@ -58,23 +57,21 @@ def __init__(self, cost, constraints, substitutions=None, verbosity=1):
     a dummy variable z to be greater than the desired Signomial objective s
     (z >= s) and then minimizing that dummy variable.""")
         CostedConstraintSet.__init__(self, cost, constraints, substitutions)
-        try:
-            self.__parse_externalfnvars()
-            if self.externalfn_vars:  # not a GP! Skip to the `except`
-                raise InvalidGPConstraint("some variables have externalfns")
-            _ = self.as_posyslt1(substitutions)
-        except InvalidGPConstraint:
-            pass
-        else:  # this is a GP
-            raise ValueError("""Model valid as a Geometric Program.
+        self.externalfn_vars = frozenset(Variable(newvariable=False, **v.descr)
+                                         for v in self.varkeys if v.externalfn)
+        self.is_sgp = bool(self.externalfn_vars)
```

Comment:
yeah, this flag really means: "a non-SP SGP". Considering the class is still called SignnomialProgram, several things probably need to be renamed.